### PR TITLE
feat(RowPinning): Initial implementation of jump to disabled row grid action.

### PR DESCRIPTION
**Pull request by skrustev**
#6941

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by MayaKirova** [comment]
@skrustev Build fails with error:
Property '_pinnedRecordIDs' does not exist on type 'IgxGridBaseDirective'

Additionally pinning throws error in treegrid: 
Cannot set property 'ghostIndex' of undefined

File: projects/igniteui-angular/src/lib/grids/common/pipes.ts [hunk 1]
```diff
@@ -229,7 +229,7 @@ export class IgxGridRowPinningPipe implements PipeTransform {
 
         if (grid.hasPinnedRecords && isPinned) {
             const result = collection.filter(rec => grid.isRecordPinned(rec));
-            result.sort((rec1, rec2) => grid.pinRecordIndex(rec1) - grid.pinRecordIndex(rec2));
+            result.sort((rec1, rec2) => grid.getPinedRecordIndex(rec1) - grid.getPinedRecordIndex(rec2));
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/common/pipes.ts:232]: getPinedRecordIndex and setPinnedRecordGhostIndex will probably have to be overwritten for the treegrid as its data records are of different type - ITreeGridRecord. Hence these methods will not work as expected for them.

**Comment by MayaKirova** [comment]
Issues:
- [ ] Pinned rows order is not retained correctly in TreeGrid.
 To reproduce pin the second row in tree grid and then the first row. Order in pinned area should be second row and then first row since this is the order of pinning.

**Comment by skrustev** [comment]
This PR should be merged after #7192, since it contains changes that are required for the Tree Grid Jump To on another page to work.

File: projects/igniteui-angular/src/lib/action-strip/grid-actions/grid-pinning-actions.component.html [hunk 2]
```diff
@@ -1,4 +1,7 @@
 <ng-container *ngIf="isRowContext">
+    <button *ngIf="inPinnedArea" igxRipple igxButton="icon" (click)="scrollToRow($event)">
+        <igx-icon fontSet="filtering-icons" name="jump_down"></igx-icon>
```
Comment (dkamburov) [inline projects/igniteui-angular/src/lib/action-strip/grid-actions/grid-pinning-actions.component.html:3]: There should be jump_up icon when the rows are pinned to bottom

**Comment by MayaKirova** [comment]
Hierarchical Grid:
- [ ] In root grid if a row in the middle is pinned and all rows are then expanded, "Go To" cannot scroll to the correct row.

Steps to reproduce:
1. In the gridRowPinning sample in the hierarchical grid, pin a row in the middle (8).
2.  Expand all rows via the header button and use the "Go To" button.

- [ ] HierarchicalGrid + Paging - In HierarchicalGrid if you go from a page that has no scrollbar to a target row on a page with scrollbar the row is not scrolled in view.

Steps to reproduce:
1. In the gridRowPinning sample in the hierarchical grid set page size to 5.
2. Pin row 3 and expand a row above it (for example row 2).
3. Go to the next page and use the "Go to" button.
Page is changed but the row is not scrolled in view.

**Pull request merged**